diff --git a/debian/control b/debian/control
index ee8690e..40cd26a 100644
--- a/debian/control
+++ b/debian/control
@@ -44,6 +44,7 @@ Description: libCEC communication Library (shared library)
 Package: cec-utils
 Architecture: any
 Section: utils
+Replaces: xbian-package-cec
 Depends: libcec3 (= ${binary:Version}),
          ${shlibs:Depends},
          ${misc:Depends}
@@ -64,6 +65,8 @@ Description: Python bindings for libCEC
 
 Package: libcec
 Architecture: any
-Depends: ${shlibs:Depends}, ${misc:Depends}, libcec3 (= ${binary:Version})
+Depends: ${shlibs:Depends}, ${misc:Depends}, libcec3 (= ${binary:Version}), python-libcec, cec-utils, libcec-dev
+Provides: xbian-package-cec
+Breaks: xbian-package-cec (<<10.0)
 Description: Meta package libCEC.
 
