--- a/src/libcec/cmake/CheckPlatformSupport.cmake
+++ b/src/libcec/cmake/CheckPlatformSupport.cmake
@@ -75,9 +75,12 @@ else()
   endif()
 
   # raspberry pi
-  check_library_exists(bcm_host vchi_initialise "" HAVE_RPI_API)
+  find_library(BCM_HOST bcm_host)
+  check_library_exists(${BCM_HOST} bcm_host_init "" HAVE_RPI_API)
   if (HAVE_RPI_API)
     set(LIB_INFO "${LIB_INFO}, 'RPi'")
+    # find includes files on Raspberry Pi
+    include_directories(/opt/vc/include /opt/vc/include/interface/vcos/pthreads /opt/vc/include/interface/vmcs_host/linux)
     list(APPEND CMAKE_REQUIRED_LIBRARIES "vcos")
     list(APPEND CMAKE_REQUIRED_LIBRARIES "vchiq_arm")
     set(CEC_SOURCES_ADAPTER_RPI adapter/RPi/RPiCECAdapterDetection.cpp
--- a/src/libcec/cmake/LinkPlatformSupport.cmake
+++ b/src/libcec/cmake/LinkPlatformSupport.cmake
@@ -28,9 +28,12 @@ endif()
 
 # raspberry pi
 if (HAVE_RPI_API)
-  target_link_libraries(cec vcos)
-  target_link_libraries(cec vchiq_arm)
-  target_link_libraries(cec bcm_host)
+  find_library (VCOS vcos)
+  target_link_libraries(cec ${VCOS})
+  find_library (VCHIQ_ARM vchiq_arm)
+  target_link_libraries(cec ${VCHIP_ARM})
+  find_library (BCM_HOST bcm_host)
+  target_link_libraries(cec ${BCM_HOST})
 endif()
 
 # Apple
