commit 72c7e3b069ac18825acb63b46185958ec663fe5b
Author: Matus Kral <matuskral@me.com>
Date:   Wed Oct 8 03:56:54 2014 +0200

    don't change to bad PA (0000)

diff --git a/src/lib/adapter/IMX/IMXCECAdapterCommunication.cpp b/src/lib/adapter/IMX/IMXCECAdapterCommunication.cpp
index 0291f88..40051b4 100644
--- a/src/lib/adapter/IMX/IMXCECAdapterCommunication.cpp
+++ b/src/lib/adapter/IMX/IMXCECAdapterCommunication.cpp
@@ -202,6 +202,7 @@ cec_vendor_id CIMXCECAdapterCommunication::GetVendorId(void)
 uint16_t CIMXCECAdapterCommunication::GetPhysicalAddress(void)
 {
   uint8_t phy_addr[4];
+  uint16_t pa_tmp;
 
   if (m_dev->Ioctl(HDMICEC_IOC_GETPHYADDRESS, &phy_addr) != 0)
   {
@@ -209,7 +210,9 @@ uint16_t CIMXCECAdapterCommunication::GetPhysicalAddress(void)
     return CEC_INVALID_PHYSICAL_ADDRESS; 
   }
 
-  m_physicalAddress = ((phy_addr[0] << 4 | phy_addr[1]) << 8) | (phy_addr[2] << 4 | phy_addr[3]);
+  if ((pa_tmp = ((phy_addr[0] << 4 | phy_addr[1]) << 8) | (phy_addr[2] << 4 | phy_addr[3])))
+    m_physicalAddress = pa_tmp;
+
   return m_physicalAddress;
 }
 
